solver/src/driver.rs

```diff
@@ -125,6 +126,18 @@ impl Driver {
         }
     }
 
+    async fn can_settle(&self, settlement: RatedSettlement) -> Result<bool> {
+        let simulations = settlement_simulation::simulate_settlements(
+            chain![settlement.settlement.clone().into()],
```

Comment:
nanonit: maybe Rated settlement could become clone directly and implement Into<EncodedSettlement> to avoid this long chain.